### ISSUE TITLE
collision geometry for <link name="world"> not included in RigidBody

### ISSUE THREAD

**Issue by dmsj** [RobotLocomotion/drake#7377]
This description has been updated; the issue remains but it now references `MultibodyPlant`(whereas it originally referenced `RigidBodyTree`). Concrete code examples (and output) have been excised as not compiling against `MBP` and new comments are __marked like this__.

--------------------------------------------------

@RussTedrake per our conversation:

When creating a new ~`RigidBodyTree`~ `MultibodyPlant` for collision free IK planning, ~e.g. `self.robot = pydrake.rbtree.RigidBodyTree(self.config["ik_urdf"])`~, it appears that `<collision>` geometry in the `<link name="world">` is being incorrectly filtered out. __As well as `<visual>` geometry.__

For example, with the following URDF world definition:
```xml
  <link name="world">
    <visual>
      <origin xyz="0 0 -0.005"/>
      <geometry>
        <box size="3.0 3.0 0.01"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 -0.005"/>
      <geometry>
        <box size="3.0 3.0 0.01"/>
      </geometry>
    </collision>
  </link>
```

__The conversation below details the historical reasons for this in `RigidBodyTree` and this has propagated into the `MultibodyPlant`. As shown [here](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/detail_urdf_parser.cc#L102-L104).__

cc @mrhebert @jaymwong

**Comment by SeanCurtis-TRI** [comment]
This is a known "feature".

https://github.com/RobotLocomotion/drake/blob/master/drake/multibody/parsers/urdf_parser.cc#L500

The urdf parser explicitly skips any body named "world". I don't know the historical reason for that, but it's been that way for a long while.

**Comment by liangfok** [comment]
I believe I added the special-case handling of the "world" body here:

https://github.com/RobotLocomotion/drake/pull/2171/files#diff-414acd4947c600b5bafcc5ef751dfeadR318

The idea was to use a link called "world" to enable users to specify how the rest of the robot connects to the world, i.e., the "floating base type".

**Comment by dmsj** [comment]
Ha! I love "features." Thanks! Maybe a comment in the docs to help users down-the-line ;-)

**Comment by liangfok** [comment]
Probably the closest doc we have now is in a .cc file:

https://github.com/RobotLocomotion/drake/blob/master/drake/multibody/parsers/urdf_parser.cc#L946-L948

Though there is a need to remove the feature: #3081.

**Comment by jwnimmer-tri** [comment]
@SeanCurtis-TRI This appears to be a request specifically related to RigidBodyTree.  Can we close it as "WONTFIX" in that case?

**Comment by SeanCurtis-TRI** [comment]
We can close it.

However, my recollection is that the RBT URDF parser was ported pretty directly over for MBP. As such, I believe we have the same artifact. See [`detail_urdef_parser.cc`](https://github.com/RobotLocomotion/drake/blob/master/multibody/parsing/detail_urdf_parser.cc#L102-L104). If the body's name is "world", we skip it.

So, the issue has propagated to MBP.

**Comment by jwnimmer-tri** [comment]
Maybe update the title / overview, in that case?

**Comment by SeanCurtis-TRI** [comment]
Done

**Comment by SeanCurtis-TRI** [comment]
I believe this was inadvertantly duplicated by #14727  and resolved by #15763.
